enterprise/backend/src/metabase_enterprise/representations/v0/mbql.clj

```diff
@@ -0,0 +1,136 @@
+(ns metabase-enterprise.representations.v0.mbql
+  "MBQL transformations for import/export of representations."
+  (:require
+   [clojure.walk :as walk]
+   [metabase-enterprise.representations.lookup :as lookup]
+   [metabase-enterprise.representations.v0.common :as v0-common]
+   [metabase.lib-be.core :as lib-be]
+   [metabase.lib.core :as lib]
+   [metabase.lib.metadata :as lib.metadata]
+   [metabase.lib.normalize :as lib.normalize]
+   [metabase.lib.query :as lib.query]
+   [metabase.util :as u]
+   [metabase.util.malli :as mu]))
+
+(set! *warn-on-reflection* true)
+
+;; ============================================================================
+;; Import: Convert representation refs to Metabase IDs
+;; ============================================================================
+
+(defn- resolve-source-card
+  [card-ref ref-index _mp]
+  (when (v0-common/ref? card-ref)
+    (-> ref-index
+        (v0-common/lookup-id card-ref)
+        (v0-common/ensure-not-nil))))
+
+(defn- resolve-source-table
+  "Resolves refs in the raw MBQL map, replacing them with IDs for import."
+  [source-table _ref-index mp]
+  (when (v0-common/table-ref? source-table)
+    (->> (lib.metadata/tables mp)
+         (filter #(and (= (:schema source-table) (:schema %))
+                       (= (:table  source-table) (:name   %))))
+         first
+         :id
+         v0-common/ensure-not-nil)))
+
+(defn- resolve-field
+  [field-ref ref-index mp]
+  (when (v0-common/field-ref? field-ref)
+    (let [table-id (resolve-source-table (dissoc field-ref :field) ref-index mp)]
+      (->> (lib.metadata/fields mp table-id)
+           (filter #(= (:field field-ref) (:name %)))
+           first
+           :id
+           v0-common/ensure-not-nil))))
+
+(defrecord V0ImportVisitor []
```

Comment:
Opened DevEx issue for this https://linear.app/metabase/issue/DEV-1129/add-kondo-linter-config-for-preferring-pdefrecord-and-pdefprotocol